src/app/js/formats/heatmap/HeatMapView.js

```diff
@@ -0,0 +1,224 @@
+        display: 'block',
+        height: '30px',
+        margin: '2px',
+        lineHeight: '30px',
+    },
+    rotate: {
+        position: 'relative',
+        top: '0px',
+        left: '0px',
+        height: '100%',
+        transform: 'skew(-45deg,0deg)',
+    },
+    rotateSpan: {
+        transform: 'skew(45deg,0deg) rotate(315deg)',
+        position: 'absolute',
+        bottom: '30px',
+        left: '0px',
+        display: 'inline-block',
+        width: '85px',
+        textAlign: 'left',
+        whiteSpace: 'normal',
+    },
+});
+
+const getColorStyle = color => ({
+    backgroundColor: color,
+});
+
+class HeatMapView extends Component {
+    handleMove = event => {
+        const x = event.clientX;
+        const y = event.clientY + window.pageYOffset;
+        const value = Number(get(event, ['target', 'dataset', 'value'], null));
+
+        if (!value) {
+            this.props.hideTooltip();
+            return;
+        }
+
+        this.props.showTooltip({
+            origin: { x, y },
+            content: <p>{value}</p>,
+        });
+    };
+
+    handleLeave = () => {
+        this.props.hideTooltip();
+    };
+    render() {
+        const { xAxis, yAxis, dictionary, maxValue, colorScheme } = this.props;
+        const getColor = scaleQuantize()
+            .range(colorScheme)
+            .domain([0, maxValue])
+            .nice();
+
+        return (
+            <div>
+                <table
+                    onMouseMove={this.handleMove}
+                    onMouseLeave={this.handleLeave}
+                    className={css(styles.table)}
+                >
+                    <thead className={css(styles.thead)}>
+                        <tr className={css(styles.tr)}>
+                            <th className={css(styles.th)} />
+                            {yAxis.map(yKey => (
+                                <th className={css(styles.th)} key={yKey}>
+                                    {yKey}
+                                </th>
+                            ))}
+                        </tr>
+                    </thead>
+                    <tbody className={css(styles.tbody)}>
+                        {xAxis.map(xKey => (
+                            <tr key={xKey} className={css(styles.tr)}>
+                                <td className={css(styles.td)}>
+                                    <div className={css(styles.rotate)}>
+                                        <span
+                                            className={css(styles.rotateSpan)}
+                                        >
+                                            {xKey}
+                                        </span>
+                                    </div>
+                                </td>
+                                {yAxis.map(yKey => (
+                                    <td
+                                        className={css(styles.td)}
+                                        key={yKey}
```

Comment:
not good, you'll end up with duplicate keys as it's repeated for every line. I advise you to use a compound of `xKey` and `yKey` here

Reply:
I thought that as long as all the key in the array of component are unique, it's good, even if there is an element with the same key elsewhere. Also react print no duplicate key error.
I guess it cannot hurt, just not sure it is really needed.